src/overlays/actors/ovl_Dm_Ravine/z_dm_ravine.c

```diff
@@ -19,16 +25,55 @@ const ActorInit Dm_Ravine_InitVars = {
     (ActorFunc)DmRavine_Init,
     (ActorFunc)DmRavine_Destroy,
     (ActorFunc)DmRavine_Update,
-    (ActorFunc)DmRavine_Draw
+    (ActorFunc)DmRavine_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Init.asm")
+void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmRavine* this = THIS;
+    u8 flag = gSaveContext.perm.weekEventReg[0];
+    if (((flag & 0x10) | cREG(0)) != 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->loaded = false;
+    globalCtx->roomContext.unk7A[0] = 1;
+    globalCtx->roomContext.unk7A[1] = 0;
+    this->state = 0;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = DmRavine_DoNothing;
+}
+
+void DmRavine_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Destroy.asm")
+void DmRavine_DoNothing(DmRavine* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/func_80A2E838.asm")
+void DmRavine_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmRavine* this = THIS;
+    RoomContext* roomCtx;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Update.asm")
+    switch (this->state) {
+        case 0:
```

Comment:
These states may make sense as an enum. State 0 is doing nothing. State 1 is waiting for the cutscene to hit a specific stage, then moving to state 2 which sets the actor to be removed.